game.js

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-use-before-define */
+
+// module to create board
+const gameBoard = (() => {
+  const board = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+  return {
+    board,
+  };
+})();
+
+// factory to create players
+const Player = (name, symbol) => {
+  const getName = () => name;
+  const getSymbol = () => symbol;
+  const comb = [];
+  return { getName, getSymbol, comb };
+};
+
+// module containing all the game logic
+const gameLogic = (() => {
+  let player1;
+  let player2;
+  let currentPlayer;
+  let turns = 0;
+  const winningMoves = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6]];
+
+  // DOM manipulation for when the game ends
+  const winner = (player, draw = false) => {
+    const message = document.getElementById('message');
+    while (message.firstChild) {
+      message.removeChild(message.firstChild);
+    }
+    const messageContainer = document.createElement('div');
+    const messageHeader = document.createElement('h3');
+    const messageText = document.createElement('p');
+    if (draw === true) {
+      messageHeader.innerHTML = "It's a draw!";
+    } else {
+      messageHeader.innerHTML = `Congratulation ${player.getName()} you win!`;
+    }
+    messageText.innerHTML = 'would you like to play again?';
+    messageContainer.append(messageHeader);
+    messageContainer.append(messageText);
+    const buttonDiv = document.createElement('div');
+    buttonDiv.innerHTML = '<button id="reset_button">Play again!</button>';
+    message.append(messageContainer);
+    message.append(buttonDiv);
+    message.attributes.class.value = 'message_section';
+    game.attributes.class.value = 'none';
+    game_info.attributes.class.value = 'none';
+    document.querySelector('button').addEventListener('click', reset);
+  };
```

Comment:
### Separate app dom manipulation from app logic

Reply:
Hi Raymond,

Thank you very much for the great feedback!

have a great day,
Michael.